README.md

```diff
@@ -1,48 +1,162 @@
 # feedback-DT
 
-## Creating a BabyAI dataset
-Currently, this can be run with three different parameters. The plan is to also add an RGB parameter.
-```src/_datsets.py --env_name --num_episodes --include_timeout```
+This repository houses all the code for our paper '*Is Feedback All You Need? Leveraging Natural Language Feedback in Goal-Conditioned Reinforcement Learning*', as well as ongoing work developing the project further. If you use any of the code for your sown work, please cite the paper as
+```
+citation
+```
 
-## Passing different environment name strings
-The names of BabyAI environments can be found on this page https://minigrid.farama.org/environments/babyai/
-under "Registered Configurations". 
-A list of environments for the single-room ```goto``` tasks (to be used for prelimary experiments as per project proposal) can be found in ```envs.json```.
+# Overview and usage
+The main purpose of this codebase is to train and evaluate feedback-augmented decision-transformer agents on an extended version of the [BabyAI testbed](https://github.com/mila-iqia/babyai/tree/master).
 
-### Passing different numbers of episodes
-Note that when passing any ```--num_episodes``` with ```--include_timeout=False```, this will likely result in significantly fewer episodes than passed (due to only successful episodes being registered).
+## Installation
+You can clone the repository and install dependencies with
+```bash
+git clone https://github.com/maxtaylordavies/feedback-DT.git
+cd feedback-DT
+pipenv install
+```
 
-### Recording episodes that timed out / were truncated
-By default, the ```--include_timeout``` parameter will be parsed with ```True```, resulting in episodes that ended because they timed out (reached ```max_steps```) before being terminated by the agent. To only record successful episodes, pass ```False```
+## Training and evaluating an FDT agent
+The script `train_agent_babyai.py` contains everything needed to 
+- generate a training dataset
+- train an FDT agent
+- run periodic evaluation during training and save results
 
-### Example
-```sh
-python src/_datasets.py --env_name "BabyAI-GoToRedBallGrey-v0" --num_episodes 1000 --include_timeout False
+To run it with default arguments, simply execute 
+```bash
+python scripts/train_agent_babyai.py
 ```
 
-## Generating language feedback for an existing BabyAI dataset
+The script also supports a number of arguments. The main ones are highlighted below; for a complete list refer to `src/utils/argparsing.py`.
+
+### Specifying a BabyAI level
+The `--level` argument determines which BabyAI level to use, defaulting to "GoToRedBallGrey". A list of levels can be found on [this page](https://minigrid.farama.org/environments/babyai/) under "Registered Configurations". 
```

Comment:
Can we make a note that the format for specifying the level is just the middle part of the config name? e.g., "BabyAI-GoToRedBallGrey-v0" -> "GoToRedBallGrey"

Reply:
good idea